src/rgw/rgw_file.h

```diff
@@ -802,7 +802,10 @@ namespace rgw {
 			     const uint32_t flags = RGWFileHandle::FLAG_NONE) {
       using std::get;
 
-      LookupFHResult fhr { nullptr, RGWFileHandle::FLAG_NONE };
+      // This seems like a Clang compiler error.
+      // It result in irgw::RGWFileHandle::FLAG_NONE being undefined during linking.
+      // LookupFHResult fhr { nullptr, RGWFileHandle::FLAG_NONE };
+      LookupFHResult fhr { nullptr, 0 };
```

Comment:
On 23-5-2016 04:18, Kefu Chai wrote:
> In src/rgw/rgw_file.h
> <https://github.com/ceph/ceph/pull/9178#discussion_r64162856>:
> 
>> @@ -802,7 +802,10 @@ namespace rgw {
>>  			     const uint32_t flags = RGWFileHandle::FLAG_NONE) {
>>        using std::get;
>>  
>> -      LookupFHResult fhr { nullptr, RGWFileHandle::FLAG_NONE };
>> +      // This seems like a Clang compiler error.
>> +      // It result in irgw::RGWFileHandle::FLAG_NONE being undefined during linking.
>> +      // LookupFHResult fhr { nullptr, RGWFileHandle::FLAG_NONE };
>> +      LookupFHResult fhr { nullptr, 0 };
> 
>     But what I do not understand is that there are more references to
>     RGWFileHandle::FLAG_NONE ( for instance a few lines earlier ),
> 
> yeah because |constexp| /and/ passing by rvalue.
> 
>     And per your link-text, that static declaration would require an
>     explicit definition.
>     But that sort of surpasses the constexpr declaration.
> 
> please note the difference of ODR-use and non ODR-use.
> 
>     Interested in what your suggestion for a fix is.
> 
> - LookupFHResult fhr { nullptr, RGWFileHandle::FLAG_NONE };
> + LookupFHResult fhr { nullptr, uint32_t(RGWFileHandle::FLAG_NONE) };

Cool, I'll give it a try.

--WjW